### PR TITLE
meson: Require Frida fork of V8

### PR DESCRIPTION
**Pull request by milahu**
part of https://github.com/frida/frida-core/pull/452

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -529,10 +530,23 @@ if have_gumjs
     quickjs_dep_native = dependency('quickjs', native: true)
   endif
 
-  v8_dep = dependency('v8-10.0',
-    version: '>=10.6.122',
-    required: get_option('v8'),
+  # workaround for https://github.com/mesonbuild/meson/issues/11381
+  if get_option('v8').auto()
+    error('option "v8" cannot be "auto", must be either "enabled" or "disabled"')
+  endif
```
Comment (oleavr) [inline meson.build:536]: I take it that this is an outdated solution?

**Review by oleavr — CHANGES_REQUESTED** [review]

File: meson.build (hunk 1 above)
Comment (milahu) [reply meson.build]: fixed

File: meson.build [hunk 2]
```diff
@@ -529,14 +530,22 @@ if have_gumjs
...
+    not_found_message: 'please use v8 from https://github.com/frida/v8'
   )
-  cdata.set('HAVE_V8', v8_dep.found())
   if v8_dep.found()
+    assert(
+      cppc.has_header_symbol('v8-platform.h', 'v8::ThreadingBackend', dependencies: [v8_dep]),
+      'please use v8 from https://github.com/frida/v8'
+    )
+    # TODO use the actual name of the found v8_dep
     gumjs_extra_requires += 'v8-10.0'
```
Comment (milahu) [inline meson.build:546]: currently frida-gum does not work with vanilla v8, so this can be ignored

Comment (oleavr) [reply meson.build:546]: Right! I'm not sure there's much point in looking for `v8` though. But the `not_found_message` change alone does seem like a good idea.

Comment (milahu) [reply meson.build]: fixed + check for v8::ThreadingBackend

**Review by oleavr — CHANGES_REQUESTED** [review]
Thanks!

**Comment by oleavr** [comment]
Had to revert this as it caused regressions, e.g. [macos-x86_64](https://github.com/frida/frida/actions/runs/4199451005/jobs/7284379145).

**Pull request merged**